plugins/modules/ec2_vpc_route_table.py

```diff
@@ -510,29 +538,63 @@ def ensure_subnet_associations(connection=None, module=None, route_table=None, s
         new_association_ids.append(result['association_id'])
 
     if purge_subnets:
-        to_delete = [a_id for a_id in current_association_ids
-                     if a_id not in new_association_ids]
+        to_delete = [association_id for association_id in current_association_ids
+                     if association_id not in new_association_ids]
 
-        for a_id in to_delete:
+        for association_id in to_delete:
             changed = True
             if not check_mode:
                 try:
-                    connection.disassociate_route_table(aws_retry=True, AssociationId=a_id)
+                    connection.disassociate_route_table(aws_retry=True, AssociationId=association_id)
                 except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
                     module.fail_json_aws(e, msg="Couldn't disassociate subnet from route table")
 
     return {'changed': changed}
 
 
+def ensure_gateway_association(connection=None, module=None, route_table=None, gateway_id=None,
+                               check_mode=None):
+    changed = False
+    filters = ansible_dict_to_boto3_filter_list({'association.gateway-id': gateway_id, 'vpc-id': route_table['VpcId']})
+    try:
+        route_tables = describe_route_tables_with_backoff(connection, Filters=filters)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get route tables")
+    for table in route_tables:
+        if table['RouteTableId'] is None:
+            continue
+        for association in table['Associations']:
+            if association['Main']:
+                continue
+            if association['GatewayId'] == gateway_id:
+                if table['RouteTableId'] == route_table['RouteTableId']:
+                    return {'changed': False, 'association_id': association['RouteTableAssociationId']}
+                if check_mode:
+                    return {'changed': True}
+                try:
+                    connection.disassociate_route_table(
+                        aws_retry=True, AssociationId=association['RouteTableAssociationId'])
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    module.fail_json_aws(e, msg="Couldn't disassociate gateway from route table")
+
+    try:
```

Comment:
Im not quite sure the best way to handle dissociating a gateway from a route table. i essentially replicated the `ensure_subnet_associations` method, but the way that is done, subnets are dissociated when `subnets = []` is specified. Cant do the same for a gateway. @jillr @alinabuzachis

Reply:
Not sure if you mean for `check_mode` or in general. If for `check_mode`, I left a comment.

Reply:
in general - the check_mode case seems pretty trivial. Ill put something together for it I just wasnt sure if there were other modules I could use as a reference!

Reply:
updated - only slight difference from disassociating a subnet is after disassociating a gateway, it lingers around with a state of "disassociated" rather than being removed, so i had to adjust the integration tests a little